backend/pkg/k8cache/k8cache.go

```diff
@@ -195,3 +195,15 @@ func SetHeader(cacheData CachedResponseData, w http.ResponseWriter) {
 	w.Header().Set("X-HEADLAMP-CACHE", "true")
 	w.WriteHeader(cacheData.StatusCode)
 }
+
+// MarshallToStore serialize a cacheResponseData struct into JSON []byte.
+// This function is used before storing the K8's response data into cache.
+// ensuring a consistent and structured format for all cached entries.
+func MarshalToStore(cacheData CachedResponseData) ([]byte, error) {
```

Comment:
I don't think this function is needed?

Seems we don't gain anything from it compared to just calling json.Marshal.

jsonByte, err := json.Marshal(cacheData)
jsonByte, err := MarshalToStore(cacheData)